src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.cpp

```diff
@@ -0,0 +1,443 @@
+                                        req.caid.Value(), [&](auto & certificate) -> CHIP_ERROR {
+                                            FindRootCertificateResponse::Type response;
+                                            DataModel::List<const TLSCertStruct::Type> details(&certificate, 1);
+                                            response.certificateDetails = details;
+                                            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                            return CHIP_NO_ERROR;
+                                        });
+    }
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleLookupRootCertificate(HandlerContext & ctx,
+                                                                 const LookupRootCertificate::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: LookupRootCertificate");
+    auto result = mDelegate.LookupRootCertByFingerprint(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(),
+                                                        req.fingerprint, [&](auto & certificate) -> CHIP_ERROR {
+                                                            LookupRootCertificateResponse::Type response;
+                                                            response.caid = certificate.caid;
+                                                            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+                                                            return CHIP_NO_ERROR;
+                                                        });
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    }
+}
+
+void TlsCertificateManagementServer::HandleRemoveRootCertificate(HandlerContext & ctx,
+                                                                 const RemoveRootCertificate::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsCertificateManagement: RemoveRootCertificate");
+    auto result = mDelegate.RemoveRootCert(ctx.mRequestPath.mEndpointId, ctx.mCommandHandler.GetAccessingFabricIndex(), req.caid);
+    if (result == CHIP_ERROR_NOT_FOUND)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+    }
+    else if (result == CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Success);
+    }
+    else
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
```

Comment:
Spec has 

>  Fail the command with the status code INVALID_IN_STATE, and end processing with no other side effects.

in some situations here.  How is that supposed to happen?

It really might have been clearer if either more logic lived in the cluster or if the delegate callbacks returned status, if we're supposed to end up with a status in various cases....

Reply:
Apologies, this was supposed to use `Protocols::InteractionModel::Status` to defer to app, updated